KubeArmor/core/kubeUpdate.go

```diff
@@ -516,13 +516,20 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 				endpoints = append(newendpoints, endpoint)
 			}
 
-			dm.EndPointsLock.Lock()
+			dm.EndPointsLock.RLock()
+			endPointsLength := len(dm.EndPoints)
+			dm.EndPointsLock.RUnlock()
 
 			idx := 0
 			nidx := 0
-			for nidx < len(endpoints) && idx < len(dm.EndPoints) {
-				if pod.Metadata["namespaceName"] == dm.EndPoints[idx].NamespaceName && pod.Metadata["podName"] == dm.EndPoints[idx].EndPointName && kl.ContainsElement(endpoints, dm.EndPoints[idx].ContainerName) {
+			for nidx < len(endpoints) && idx < endPointsLength {
+				dm.EndPointsLock.RLock()
+				endPoint := dm.EndPoints[idx]
+				dm.EndPointsLock.RUnlock()
+				if pod.Metadata["namespaceName"] == endPoint.NamespaceName && pod.Metadata["podName"] == endPoint.EndPointName && kl.ContainsElement(endpoints, endPoint.ContainerName) {
+					dm.EndPointsLock.Lock()
 					dm.EndPoints[idx] = endpoints[nidx]
+					dm.EndPointsLock.Unlock()
```

Comment:
Let's not do this, it's unnecessary. It's fine to have the lock outside the loop and keep it like earlier.